source/libip6string/ip6tos.c

```diff
@@ -92,4 +95,27 @@ void ip6tos(const void *ip6addr, char *p)
         }
     }
     *p++ = '\0';
+
+    // Return length of generated string, excluding the terminating null-character
+    return p - p_orig - 1;
+}
+
+uint8_t ip6_prefix_tos(const uint8_t *prefix, uint8_t prefix_len, char *p)
+{
+    char *wptr = p;
+    uint8_t addr[16] = {0};
+
+    if (((prefix_len != 0) && (prefix == NULL )) || prefix_len > 128) {
+        strcpy(p, "<err>");
+        return 5;
+    }
+
+    // Generate prefix part of the string
+    bitcopy(addr, prefix, prefix_len);
+    wptr += ip6tos(addr, wptr);
+    // Add the prefix length part of the string
+    wptr += sprintf(wptr, "/%u", prefix_len);
```

Comment:
If you do change to uint_fast8_t, this should be "/%"PRIuFAST8. Technically I guess it should be "/%"PRIu8 now.